Script/AtomicNET/AtomicNETService/CSComponentInspector.cs

```diff
@@ -34,721 +34,796 @@
 namespace AtomicTools
 {
 
-	public class CSComponentInspector
-	{
+    public class CSComponentInspector
+    {
 
-		InspectorComponent _inspectorComponent;
+        InspectorComponent _inspectorComponent;
 
-		public CSComponentInspector (TypeDefinition typeDef, PEReader peFile, MetadataReader metaReader)
-		{
-			this.typeDef = typeDef;
-			this.peFile = peFile;
-			this.metaReader = metaReader;
+        public CSComponentInspector(TypeDefinition typeDef, PEReader peFile, MetadataReader metaReader)
+        {
+            this.typeDef = typeDef;
+            this.peFile = peFile;
+            this.metaReader = metaReader;
 
-			this._inspectorComponent = new InspectorComponent ();
-			this._inspectorComponent.Name = metaReader.GetString (typeDef.Name);
-			this._inspectorComponent.Namespace = metaReader.GetString (typeDef.Namespace);
-		}
+            this._inspectorComponent = new InspectorComponent();
+            this._inspectorComponent.Name = metaReader.GetString(typeDef.Name);
+            this._inspectorComponent.Namespace = metaReader.GetString(typeDef.Namespace);
+        }
 
-		// Inspect a CSComponent derived class
-		internal InspectorComponent Inspect ()
-		{
+        // Inspect a CSComponent derived class
+        internal InspectorComponent Inspect()
+        {
+            var baseTypeHandle = typeDef.BaseType;
 
-			var fields = typeDef.GetFields ();
+            // Inspect parents
+            while (baseTypeHandle.Kind == HandleKind.TypeDefinition)
```

Comment:
This is the change. Iterate over parents, inspecting their fields. Should know better than to mix a whitespace correct and other fixes.